crypto/crmf/crmf_lib.c

```diff
@@ -629,7 +627,7 @@ X509 *OSSL_CRMF_ENCRYPTEDVALUE_get1_encCert(const OSSL_CRMF_ENCRYPTEDVALUE *ecer
     }
     cikeysize = EVP_CIPHER_key_length(cipher);
     /* first the symmetric key needs to be decrypted */
-    pkctx = EVP_PKEY_CTX_new(pkey, NULL);
+    pkctx = EVP_PKEY_CTX_new_from_pkey(libctx, pkey, propq);
```

Comment:
I would like to see a X509_new_with_libctx() used here which is then passed to d2i_X509

Reply:
> I would like to see a X509_new_with_libctx() used here which is then passed to d2i_X509

I don't get what you mean here.
This change is not about `X509` but `EVP_PKEY_CTX`.
If any change is actually needed here please let me know the concrete code.

Reply:
Err I cant mark the line but look at line 686 which should be
```
cert = X509_new_with_libctx(libctx, propq);
if (cert == NULL) {
}
if (d2i_X509(&cert, &p, outlen) == NULL) {
```

Reply:
This will do the cache_extensions as I have described above..

Reply:
Ah, so your comment was referring not to the right position in the code.
Thanks for the code suggestion; I've used it (in a more compact form).